### PR TITLE
feat: Add KYC flow

### PR DESCRIPTION
**Pull request by lszetg**
## Motivation and Context

Add KYC flow with new views, widgets and Riverpod-based model

## Summary

This PR introduces the full KYC process. It includes:
- new screens for each KYC step
- shared widgets (form inputs, document previews, progress bar, etc.)
- a KycForm model using Riverpod for state management
- validation logic for each step
- mobile and web image support (file upload vs. memory)

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines (see `.cursorrules` and `AGENTS.md`).
- [ ] No redundant or duplicate code.
- [ ] No new dependencies added without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments, AGENTS.md).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases (unit/widget/integration as appropriate).
- [ ] Tests pass locally (`flutter test`).

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues (see GitHub Actions).
- [ ] Updated build scripts or workflows if necessary.

## AI Agent Guidance

_If this PR was created by an AI agent:_

- [ ] Clearly state the user prompt or intent that triggered this PR.
- [ ] Reference any relevant files, guidelines, or previous PRs/issues.
- [ ] Ensure all code changes are explained and justified in the summary.
- [ ] If unsure about a decision, leave a comment for human reviewers to verify.

## Comments to Reviewer

_Provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process._

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

**Review by mighty840 — APPROVED** [review]
In general, this looks quite good. Some improvements can be made, but I think for now we dont need to optimize this ultra. We need to anyways wait for the compliance guidelines to refine the onboarding flow and parameters. So far, this looks in line with our design, so let's go :rocket:

File: lib/screens/onboarding_screen.dart [hunk 1]
```diff
@@ -222,4 +222,4 @@ class _OnboardingScreenState extends State<OnboardingScreen> {
       ),
     );
   }
-}
+}
```
[the remark this answers is not in the document]
Comment (lszetg) [reply lib/screens/onboarding_screen.dart:225]: 😆 
tsc
